src/Microsoft.VisualStudio.Services.Agent/Logging.cs

```diff
@@ -36,12 +56,54 @@ public override void Initialize(IHostContext hostContext)
         //
         public void Write(string message)
         {
-            if (TimeLineId == Guid.Empty)
+            ArgUtil.NotEmpty(_timelineId, nameof(_timelineId));
```

Comment:
i wouldnt add any arg validation. better to keep this fast since it's handling output spewing from the build.